sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusTriggerMetrics.cs

```diff
@@ -8,22 +8,25 @@
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Listeners
 {
-    internal class ServiceBusTriggerMetrics : ScaleMetrics
+    /// <summary>
+    /// Trigger metrics class for Service Bus. Used to make scaling decisions in <see cref="ServiceBusScaleMonitor"/>.
+    /// </summary>
+    public class ServiceBusTriggerMetrics : ScaleMetrics
     {
         /// <summary>
         /// The number of messages currently in the queue/topic.
         /// </summary>
-        public long MessageCount { get; set; }
+        public long MessageCount { get; internal set; }
```

Comment:
I think the setters should be public. Kind of strange (and not testable) to have a public data type like this that you can instantiate but not provide values for. Also note that the base class property has a public setter.

Reply:
Normally what we do for output types like this is make the constructor internal and provide a model factory for testing. I missed that the base class already has a setter, so I would agree that we can just revert this change.